test/e2e/framework/repository.go

```diff
@@ -103,9 +105,38 @@ func (f *Framework) BackupCountInRepositoriesStatus(repos []*api.Repository) int
 
 	// use minimum backupCount among all repos
 	for _, repo := range repos {
-		if repo.Status.BackupCount < backupCount {
-			backupCount = repo.Status.BackupCount
+		if int64(repo.Status.BackupCount) < backupCount {
+			backupCount = int64(repo.Status.BackupCount)
 		}
 	}
 	return backupCount
 }
+
+func (f *Framework) CreateRepository(repo *api.Repository) error {
+	_, err := f.StashClient.StashV1alpha1().Repositories(repo.Namespace).Create(repo)
+
+	return err
+
+}
+
+func (f *Invocation) Repository(sec core.Secret, pvcName string) *api.Repository {
```

Comment:
as you are using only secret name. you can send only metadata of the secret instead of sending entire secret.